File: array/js/max-consecutive-ones.js
function getMaxConsecutiveOnesCount(arr) {
    let count = 1
    let maxCount = 1

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] === arr[i-1]) {
            count++
            maxCount = Math.max(maxCount, count)
        } else {
            count = 1
        }
    }

    console.log("Max consecutive ones count:", maxCount)
    return maxCount
}

// let arr = [1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 1, 1]
// let arr = [0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1]
let arr = [0, 0, 0, 0]
getMaxConsecutiveOnesCount(arr)

File: array/rust/src/max_consecutive_ones.rs
pub fn get_max_consecutive_ones_count(arr: &[i32]) -> Option<i32> {
    if arr.is_empty() {
        return None;
    };

    let mut max_count = 1;
    let mut count = 1;

    for window in arr.windows(2) {
        if window[0] == window[1] {
            count += 1;
            max_count = max_count.max(count);
        } else {
            count = 1;
        }
    }

    println!("Max consecutive ones count: {}", max_count);
    Some(max_count)
}
